### ISSUE TITLE
Add virtiofs to `tmt provision` local qemu

### ISSUE THREAD

**Issue by cgwalters** [teemtee/tmt#2348]
See https://dustymabe.com/2023/09/08/using-virtiofs-with-libvirt/virt-install/

Basically virtiofs is really really useful for interacting with local guests as it makes getting data in and out not completely suck.  

It looks like this is really a rfe for https://pagure.io/testcloud but because this project wraps it it'd need to be plumbed through here too.

(This may actually turn into a bit of unprivileged-libvirt virtiofsd work, similar to what I did in coreos-assembler)

**Comment by thrix** [comment]
Nice. I assume this would be an alternative to copying data over network and "share" it similarly as we do with podman, thus speeding up the test data synchronization.

**Comment by cgwalters** [comment]
Yes, it's dramatically more ergonomic and efficient for interacting with local VM than copying files over SSH.

**Comment by frantisekz** [comment]
testcloud maint here, I see "Currently the virtiofs integration doesn’t work as non-root via a qemu:///session connection." in the linked blogpost, which would be a problem since tmt uses user session in the most cases (by default).

**Comment by cgwalters** [comment]
> Currently the virtiofs integration doesn’t work as non-root via a qemu:///session connection." 

Right, that's on track to be fixed https://gitlab.com/libvirt/libvirt/-/issues/535#note_1549107870

Anyways actually the immediate thing I'm trying to do is use `tmt provision`/`testcloud` as a "development shell" provisioner, I'd actually be OK with using the system libvirt too in some cases.

**Comment by frantisekz** [comment]
This is now WIP testcloud side: https://pagure.io/testcloud/pull-request/160#

**Comment by psss** [comment]
Seems that the `testcloud` part has been implemented:

* https://github.com/teemtee/testcloud/commit/c902d0a

So, I guess, we should be unblocked here. @cgwalters or @frantisekz, would you be interested in preparing the patch?

**Comment by psss** [comment]
Ah, sorry, just noticed there's already:

* https://github.com/teemtee/tmt/pull/3143

@frantisekz, let's finish that one?
